source/fake.go

```diff
@@ -45,18 +43,19 @@ func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
-// NewFakeSource creates a new fakeSource with the given client and namespace scope.
-func NewFakeSource(dnsName string) (Source, error) {
-	if dnsName == "" {
-		dnsName = defaultDNSName
+// NewFakeSource creates a new fakeSource with the given config.
+func NewFakeSource(cfg *Config) (Source, error) {
```

Comment:
IMO it makes sense if config struct is used to initialise the same object. In this case, for example NewFakeSource can take config.KubeMaster as a parameter ? As an end-user I would be really surprised to see this field and would not know if it is required and what is the purpose. Normally config struct fields can be omitted, but then they are there only to overwrite default values.